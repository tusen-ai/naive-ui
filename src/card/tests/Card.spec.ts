import { mount } from '@vue/test-utils'
import { NCard } from '../index'

describe('n-card', () => {
  it('should work with import on demand', () => {
    mount(NCard)
  })

  it('should work with `title` prop', async () => {
    const wrapper = mount(NCard)

    expect(wrapper.find('.n-card-header').exists()).toBe(false)

    await wrapper.setProps({ title: 'test' })
    expect(wrapper.find('.n-card-header').exists()).toBe(true)
    expect(wrapper.find('.n-card-header__main').text()).toBe('test')
  })

  it('should work with `size` prop', async () => {
    const wrapper = mount(NCard)

    await wrapper.setProps({ size: 'small' })
    expect(wrapper.find('.n-card').attributes('style')).toMatchSnapshot()
  })

  it('should work with `hoverable` prop', async () => {
    const wrapper = mount(NCard, {
      props: {
        hoverable: true
      }
    })

    expect(wrapper.find('.n-card').classes()).toContain('n-card--hoverable')
  })

  it('should work with `embedded` prop', async () => {
    const wrapper = mount(NCard)

    expect(wrapper.find('.n-card').attributes('style')).toContain(
      '--n-color: #fff'
    )
    await wrapper.setProps({ embedded: true })
    expect(wrapper.find('.n-card').attributes('style')).toContain(
      '--n-color: rgb(250, 250, 252)'
    )
  })

  it('should work with `segmented` prop', async () => {
    const wrapper = mount(NCard)
    expect(wrapper.find('.n-card').classes()).not.toContain(
      'n-card--content-segmented'
    )

    await wrapper.setProps({
      segmented: {
        content: true,
        footer: 'soft'
      }
    })
    expect(wrapper.find('.n-card').classes()).toContain(
      'n-card--content-segmented'
    )
  })

  it('should work with `slots` ', async () => {
    const wrapper = mount(NCard, {
      slots: {
        cover: () => 'cover',
        header: () => 'header',
        'header-extra': () => 'header-extra',
        default: () => 'content',
        footer: () => 'footer',
        action: () => 'action'
      }
    })

    expect(wrapper.find('.n-card-cover').exists()).toBe(true)
    expect(wrapper.find('.n-card-cover').text()).toBe('cover')

    expect(wrapper.find('.n-card-header').exists()).toBe(true)
    expect(wrapper.find('.n-card-header__main').text()).toBe('header')

    expect(wrapper.find('.n-card-header__extra').exists()).toBe(true)
    expect(wrapper.find('.n-card-header__extra').text()).toBe('header-extra')

    expect(wrapper.find('.n-card__content').exists()).toBe(true)
    expect(wrapper.find('.n-card__content').text()).toBe('content')

    expect(wrapper.find('.n-card__footer').exists()).toBe(true)
    expect(wrapper.find('.n-card__footer').text()).toBe('footer')

    expect(wrapper.find('.n-card__action').exists()).toBe(true)
    expect(wrapper.find('.n-card__action').text()).toBe('action')
  })

  it('should work with `bordered` prop', async () => {
    const wrapper = mount(NCard, {
      props: {
        bordered: false
      }
    })

    expect(wrapper.find('.n-card--bordered').exists()).toBe(false)
  })

  it('should work with `closable` and `on-close` prop', async () => {
    const onClose = vi.fn()
    const wrapper = mount(NCard, {
      props: {
        closable: true,
        onClose
      }
    })

    expect(wrapper.find('.n-card-header__close').exists()).toBe(true)
    await wrapper.find('.n-card-header__close').trigger('click')
    expect(onClose).toHaveBeenCalled()
  })

  it('should work with `header-style` prop', async () => {
    const testStyle = 'padding: 0px;'
    const wrapper = mount(NCard, {
      props: {
        title: 'test',
        headerStyle: testStyle
      },
      slots: {
        default: () => 'test'
      }
    })

    expect(wrapper.find('.n-card-header').attributes('style')).toContain(
      testStyle
    )
  })

  it('should work with `content-style` prop', async () => {
    const testStyle = 'padding: 0px;'
    const wrapper = mount(NCard, {
      props: {
        contentStyle: testStyle
      },
      slots: {
        default: () => 'test'
      }
    })

    expect(wrapper.find('.n-card__content').attributes('style')).toContain(
      testStyle
    )
  })

  it('should work with `footer-style` prop', async () => {
    const testStyle = 'padding: 0px;'
    const wrapper = mount(NCard, {
      props: {
        footerStyle: testStyle
      },
      slots: {
        footer: () => 'test'
      }
    })

    expect(wrapper.find('.n-card__footer').attributes('style')).toContain(
      testStyle
    )
  })

  it('should work with `header-extra-style` prop', async () => {
    const testStyle = 'padding: 0px;'
    const wrapper = mount(NCard, {
      props: {
        headerExtraStyle: testStyle
      },
      slots: {
        header: () => 'test-header',
        'header-extra': () => 'test-header-extra'
      }
    })

    expect(wrapper.find('.n-card-header__extra').attributes('style')).toContain(
      testStyle
    )
  })
})
