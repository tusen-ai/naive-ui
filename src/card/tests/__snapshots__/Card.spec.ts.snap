// Vitest Snapshot v1

exports[`n-card > should work with \`size\` prop 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 3px; --n-color: #fff; --n-color-modal: #fff; --n-color-popover: #fff; --n-color-target: #18a058; --n-text-color: rgb(51, 54, 57); --n-line-height: 1.6; --n-action-color: rgb(250, 250, 252); --n-title-text-color: rgb(31, 34, 37); --n-title-font-weight: 500; --n-close-icon-color: rgba(102, 102, 102, 1); --n-close-icon-color-hover: rgba(102, 102, 102, 1); --n-close-icon-color-pressed: rgba(102, 102, 102, 1); --n-close-color-hover: rgba(0, 0, 0, .09); --n-close-color-pressed: rgba(0, 0, 0, .13); --n-border-color: rgb(239, 239, 245); --n-box-shadow: 0 1px 2px -2px rgba(0, 0, 0, .08), 0 3px 6px 0 rgba(0, 0, 0, .06), 0 5px 12px 4px rgba(0, 0, 0, .04); --n-padding-top: 12px; --n-padding-bottom: 12px; --n-padding-left: 16px; --n-font-size: 14px; --n-title-font-size: 16px; --n-close-size: 22px; --n-close-icon-size: 18px; --n-close-border-radius: 3px;"`;
